doc/swish/http.tex

```diff
@@ -333,18 +479,18 @@
 exception \code{\#(bad-arg http:write-status \var{status})} is
 raised.
 
-According to HTTP~\cite{RFC7230} the status line includes a human
-readable reason phrase. The grammar shows that it can in fact be 0
-characters long; therefore, the reason phrase is not included in this
-implementation.
+According to HTTP~\cite{RFC7230} the status line includes a
+human-readable reason phrase. The grammar shows that it can in fact be
+0 characters long; therefore, the reason phrase is not included in
+this implementation.
 
 \defineentry{http:write-header}
 \begin{procedure}
   \code{(http:write-header \var{op} \var{header})}
 \end{procedure}
 \returns{} unspecified
 
-The \code{http:write-header} procedure writes the HTTP \var{header},
+The \code{http:write-header} procedure writes the HTTP \var{header}
 and trailing CR LF to the binary output port \var{op}.
 
 \var{header} is an association list. If \var{header}'s keys are not
```

Comment:
http:read-header returns a JSON object, but http:write-header takes an association list. The association list may be easier to express when writing, and the JSON is faster to query when reading.

Reply:
For consistency, consider making header a JSON object. In http:respond, note that it side-effects the JSON header by adding the keys listed in the description if they're not already present.